inception/inception-api-annotation-experimental/src/main/ts/client/AnnotationExperienceAPIImpl.ts

```diff
@@ -16,293 +16,448 @@
+import {CreateRelationResponse} from "./messages/response/relation/CreateRelationResponse";
+import {SelectRelationResponse} from "./messages/response/relation/SelectRelationResponse";
+import {SelectRelationRequest} from "./messages/request/relation/SelectRelationRequest";
+import {DeleteRelationRequest} from "./messages/request/relation/DeleteRelationRequest";
+import {UpdateRelationRequest} from "./messages/request/relation/UpdateRelationRequest";
+import {CreateRelationRequest} from "./messages/request/relation/CreateRelationRequest";
+import {DeleteRelationResponse} from "./messages/response/relation/DeleteRelationResponse";
+import {AllSpanResponse} from "./messages/response/span/AllSpanResponse";
+import {AllRelationResponse} from "./messages/response/relation/AllRelationResponse";
+import {AllSpanRequest} from "./messages/request/span/AllSpanRequest";
+import {AllRelationRequest} from "./messages/request/relation/AllRelationRequest";
+import {Viewport} from "./model/Viewport";
 
 export class AnnotationExperienceAPIImpl implements AnnotationExperienceAPI {
 
     //Websocket and stomp broker
-    stompClient: Client;
-    connected: boolean = false;
+    private _stompClient: Client;
 
     //States to remember by client
-    client: string;
-    projectID: string;
-
-    documentID: string;
-
+    private _clientName: string;
+    private _projectID: number;
+    private _documentID: number;
 
     //Text and annotations
-    text: String[];
-    annotations: Annotation[];
-    selectedAnnotation: Annotation;
+    private _text: string[];
+    private _spans: Span[];
+    private _selectedSpan: Span;
 
-    //Viewport
-    viewport: number[][];
+    private _relations: Relation[];
+    private _selectedRelation: Relation;
 
-    //Visualizer
-    readonly visualizer: AnnotationExperienceAPIVisualization;
-
-    //Actionhandler
-    readonly actionhandler: AnnotationExperienceAPIActionHandler;
+    //Viewport
+    private _viewport: Viewport;
 
     constructor() {
         this.connect();
-
-        //Visualizer
-        this.visualizer = new AnnotationExperienceAPIVisualization(this);
-
-        //ActionHandler
-        this.actionhandler = new AnnotationExperienceAPIActionHandler(this);
-
-        this.actionhandler.registerDefaultActionHandler();
     }
 
 
     //CREATE WEBSOCKET CONNECTION
     connect() {
-        if (this.connected) {
-            console.log("You are already connected")
-            return;
-        }
 
-        this.stompClient = Stomp.over(function () {
+        //TODO find better solution
+
+        this._stompClient = Stomp.over(function () {
             return new WebSocket(localStorage.getItem("url"));
         });
 
+        this._viewport = new Viewport([[0,79],[80,140],[141,189]],null);
+
         const that = this;
 
-        this.stompClient.onConnect = function (frame) {
-            that.connected = true;
+        this._stompClient.onConnect = function (frame) {
```

Comment:
How about creating a class method "onConnect" and then say `this.stopClient.onConnect = this.onConnect`? Having these long anonymous inner methods makes the code more difficult to read/understand.